action_server/src/robocorp/action_server/_robo_utils/auth.py

```diff
@@ -1,4 +1,27 @@
+import os
+from typing import Optional
+from pathlib import Path
+
+
 def generate_api_key():
     import secrets
 
     return secrets.token_urlsafe(32)
+
+
+def get_api_key(datadir: Path) -> str:
+    api_key_path = os.path.join(datadir, ".api_key")
```

Comment:
Minor: as you're dealing with a `Path`, you could use `Path` methods.

i.e.:

`api_key_path = datadir / ".api_key"`

then to read:

`api_key_path.read_text('utf-8')`

and to write:

`api_key_path.write_text(api_key, "utf-8")`